### ISSUE TITLE
Change bluetooth name instead of qdomyos-zwift

### ISSUE THREAD

**Issue by amueller000** [cagnulein/qdomyos-zwift#614]
I am using the Peloton App and have 2 Yesoul bikes in use at the same time . I only want to connect my Apple Watch and my bike to one of the bikes. Unfortunately I see 2x the same BT name (qdomyos-zwift) and cannot decide which one is the App on my iPhone. 
Changing the bluetoot name would be extremely helpful.
Thx!

**Comment by cagnulein** [comment]
thanks, @amueller000 my idea is to add a new setting for this in the advanced section. I will update this when it's ready

**Comment by amueller000** [comment]
Sounds great. Many thanks!

Perhaps there is an option to enter the bikes BT identifier #?

 

Von: Roberto Viola ***@***.***>
Antworten an: cagnulein/qdomyos-zwift ***@***.***>
Datum: Montag, 24. Januar 2022 um 08:29
An: cagnulein/qdomyos-zwift ***@***.***>
Cc: A M ***@***.***>, Mention ***@***.***>
Betreff: Re: [cagnulein/qdomyos-zwift] Change bluetooth name instead of qdomyos-zwift (Issue #614)

 

thanks, @amueller000 my idea is to add a new setting for this in the advanced section. I will update this when it's ready

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by cagnulein** [comment]
yeah maybe i can just add a 4 numbers at the end of "QZ" name to identify the phone

**Comment by amueller000** [comment]
This would be fabilous 

 

Von: Roberto Viola ***@***.***>
Antworten an: cagnulein/qdomyos-zwift ***@***.***>
Datum: Montag, 24. Januar 2022 um 08:34
An: cagnulein/qdomyos-zwift ***@***.***>
Cc: A M ***@***.***>, Mention ***@***.***>
Betreff: Re: [cagnulein/qdomyos-zwift] Change bluetooth name instead of qdomyos-zwift (Issue #614)

 

yeah maybe i can just add a 4 numbers at the end of "QZ" name to identify the phone

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by amueller000** [comment]
Hi, is it too complicated to have a name adjustment option? Thx.

**Comment by cagnulein** [comment]
@amueller000 no, it's not but i'm overwhelmed by request :) i will do, just please be patience :)
check the backlog here on github to see how many tickets and commit I do every day and remember that this is not my first job :)

**Comment by amueller000** [comment]
No problem. I was not supposed to be pushy. I was just wondering it is eventually to difficult..

Thx for your response and good luck with all the requests

 

Von: Roberto Viola ***@***.***>
Antworten an: cagnulein/qdomyos-zwift ***@***.***>
Datum: Dienstag, 8. Februar 2022 um 09:38
An: cagnulein/qdomyos-zwift ***@***.***>
Cc: A M ***@***.***>, Mention ***@***.***>
Betreff: Re: [cagnulein/qdomyos-zwift] Change bluetooth name instead of qdomyos-zwift (Issue #614)

 

@amueller000 no, it's not but i'm overwhelmed by request :) i will do, just please be patience :)
check the backlog here on github to see how many tickets and commit I do every day and remember that this is not my first job :)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by drmason789** [comment]
@cagnulein Is this referring to the "qdomyos-zwift" strings in peloton.cpp?

e.g.

```
    request.setHeader(QNetworkRequest::ContentTypeHeader, QStringLiteral("application/json"));
    request.setHeader(QNetworkRequest::UserAgentHeader, QStringLiteral("qdomyos-zwift"));
```

**Comment by cagnulein** [comment]
@drmason789 Mmm no I will leave that one as it is
Thanks

**Comment by amueller000** [comment]
Is this really for me? I assume you got a wrong address included…. ? 

 

Von: Roberto Viola ***@***.***>
Antworten an: cagnulein/qdomyos-zwift ***@***.***>
Datum: Samstag, 3. September 2022 um 07:05
An: cagnulein/qdomyos-zwift ***@***.***>
Cc: A M ***@***.***>, Mention ***@***.***>
Betreff: Re: [cagnulein/qdomyos-zwift] Change bluetooth name instead of qdomyos-zwift (Issue #614)

 

@drmason789 Mmm no I will leave that one as it is
Thanks

**Comment by cagnulein** [comment]
@amueller000 yes it's really for you :) you open this ticket on github, i'm not sending the email directly, it's github ;)

**Comment by drmason789** [comment]
Hi @amueller000, I saw this issue when browsing through old ones and it looked like something I could do for a little break.

The way I read it initially is that you want to be able to advertise your Yesoul bikes in bluetooth using different names.
However, I noticed that when I use QZ, it always advertises as DomyosBridge, and not qdomyos-zwift as you state in your messages.

I implemented the ability to add a 4 digit numeric suffix to the base name "DomyosBridge" in the advanced settings where @cagnulein suggested (I'm yet to test this or submit a PR). However, I've been wondering why you're saying the bikes appear as "qdomyos-zwift", when if I look at the history of the code that sets up the bluetooth advertising, it's been "DomyosBridge" or "dBridge" as far back as I've looked. This made me think it's something specifically to do with QZ's Peloton interaction.

As I'm donating time, I am hesitant to spend a lot trying to work things out if @cagnulein can simply tell me. :-)

**Comment by cagnulein** [comment]
@drmason789 That's because the iOS api from qt doesn't allow to change the name directly. I added a swift code in order to do it

**Comment by amueller000** [comment]
Many thanks to both of you

This is what I see when Connecting to my iPad (usually 2x the same name, when we use the 2nd bike).

I was not aware there as already an option to change the name??

 

 

Von: Roberto Viola ***@***.***>
Antworten an: cagnulein/qdomyos-zwift ***@***.***>
Datum: Samstag, 3. September 2022 um 15:14
An: cagnulein/qdomyos-zwift ***@***.***>
Cc: A M ***@***.***>, Mention ***@***.***>
Betreff: Re: [cagnulein/qdomyos-zwift] Change bluetooth name instead of qdomyos-zwift (Issue #614)

 

@drmason789 That's because the iOS api from qt doesn't allow to change the name directly. I added a swift code in order to do it

**Comment by cagnulein** [comment]
@amueller000 there isn't, it's what @drmason789 is trying to do :)

**Comment by cagnulein** [comment]
@drmason789 another point to keep in mind is that the number of chars avaiable on the advertising packet on bluetooth are limited. So we can't just add some number at the end...also the actual one must be preserved as the default. And of course the name should be passed to the swift code too (i can do it). I said this only to don't forget about this.

**Comment by drmason789** [comment]
Perhaps instead of DomyosBridge1234 (4 digits, actually in my branch I have it as DomyosBridge-1234), we could simply use a letter or number for 37 combinations:

For X in [0-9A-Z]

DomyosBridge
DomyosBridgeX 
DomyosBridge-X
 
or for even fewer characters:

QZBridgeX where X in [0-9A-Z]

Of course, for that last one, code that's currently looking for a name starting with DomyosBridge currently would need to look for QZBridge AND DomyosBridge for backward compatibility.

**Comment by cagnulein** [comment]
I like the QZ[0-9A-Z], we don't need to keep the "Domyos" prefix, but i would like to keep the actual behaviour as a standard, and only if a user will choose to have a different name, giving the possibility to change it with a different one.
I'm saying so because everytime I did a very little small change, I received a lot of email asking me why and how, so I have to keep everything very stable for everyone giving, of course, the possibility to change the default behaviour

**Comment by drmason789** [comment]
So basically:

In order to provide a consistent customer experience, the default behaviour will remain that the advertised name will be DomyosZwift.

In the advanced settings, there will be an option to use QZ suffixed with a single alphanumeric character, i.e. QZ[0-9A-Z].

e.g.
Use Alternative Bluetooth Name                            (o  )
Alternative Bluetooth Name                   QZ [1  v]  [OK]

Where QZ is a label, [1 v] is a dropdown, [OK] is a button, and (o  ) is a SwitchDelegate.

----

If we let the user change the QZ as well, then the code that looks for "DomyosBridge" to see if the connecting device is another instance of QZ won't be able to know all it needs to look for.

**Comment by cagnulein** [comment]
> If we let the user change the QZ as well, then the code that looks for "DomyosBridge" to see if the connecting device is another instance of QZ won't be able to know all it needs to look for.
> 
> @cagnulein Is this okay?

yes absolutely, that part of the code is shit :) i mean i added that part because i discoverd after I decided the name "DomyosBridge" that there is a conflict with the discovery engine on the Domyos treadmills :D so don't worry about this case :)

The important thing is that we have to manage also this in the swift code

Hah and yes for all the settings thing, I like it!

**Comment by drmason789** [comment]
@cagnulein Please let me know if some documentation for this feature is required, and where to put it.

**Comment by cagnulein** [comment]
> @cagnulein Please let me know if some documentation for this feature is required, and where to put it.

[QZ User Guide Sept 2022.docx](https://github.com/cagnulein/qdomyos-zwift/files/9545657/QZ.User.Guide.Sept.2022.docx)

we have this document that i would like to add here into the wiki of github, if you want you can add the info here. If you find a quick way to add this dock in markdown format here in github it will be awesome

**Comment by drmason789** [comment]
New text for the Advanced Settings section:

●	**Use Alternative Bluetooth Name**: Allows you to override the default name that QZ advertises itself as in Bluetooth, with QZ + an alphanumeric character. Use this it to distinguish multiple instances of QZ. Note that Zwift on Android appears to cache bluetooth device information and can continue to recognise QZ using a previous name. Using nRF Connect can clear this problem.
●	**Alternative Bluetooth Name**: Available when “Use Alternative Bluetooth Name” is enabled. Edit the alphanumeric suffix to make a distinctive name.

**Comment by amueller000** [comment]
Sounds great! 

Thanks a lot!

 

Von: David Mason ***@***.***>
Antworten an: cagnulein/qdomyos-zwift ***@***.***>
Datum: Montag, 3. Oktober 2022 um 18:16
An: cagnulein/qdomyos-zwift ***@***.***>
Cc: A M ***@***.***>, Mention ***@***.***>
Betreff: Re: [cagnulein/qdomyos-zwift] Change bluetooth name instead of qdomyos-zwift (Issue #614)

 

New text for the Advanced Settings section:

● Use Alternative Bluetooth Name: Allows you to override the default name that QZ advertises itself as in Bluetooth, with QZ + an alphanumeric character. Use this it to distinguish multiple instances of QZ. Note that Zwift on Android appears to cache bluetooth device information and can continue to recognise QZ using a previous name. Using nRF Connect can clear this problem.
● Alternative Bluetooth Name: Available when “Use Alternative Bluetooth Name” is enabled. Edit the alphanumeric suffix to make a distinctive name.
